optuna/study/study.py

```diff
@@ -806,6 +806,14 @@ def objective(trial):
         Args:
             params:
                 Parameter values to pass your objective function.
+
+                .. note::
+                    This method raises ``ValueError`` for invalid parameter values
+                    like not a number (e.g. ``np.nan``),
+                    non-positive value when parameter is log-scale,
+                    value which cannot be casted to float for int/float distribution,
+                    value which does not be included in choices for categorical distribution, etc.
```

Comment:
The error does not occur in the call of `enqueue_trial`. The error occurs in the next call of `suggest_***`. There is some discussions as to where such documentation should be written. I think it should be in `suggest_***` or `***Distribution`. (I prefer the later.)

By the way, we have [a rule to document the error](https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#exceptions).  I think we should NOT document the error when the parameter value is invalid, since it is obvious that such parameters are not acceptable. However, the obviousness is open for discussion.

Reply:
Thank you for your comment.
I agree to write such a kind of documentation in ``***Distribution``.

And sorry I had insufficient understanding for a rule to document the error.
I think error caused by not a number or values which cannot be casted to float/int is obvious, but non-positive value with log=True option and values does not be included in choices may require more discussion.

Reply:
I basically agree with the discussion. IMO, only it should be documented in the `log` argument of `***Distribution` that non-positive values are not allowed because this PR changes the behavior.

Reply:
I updated the code in response to this comment.